cmd/soroban-cli/src/commands/keys/generate.rs

```diff
@@ -49,6 +50,47 @@ pub struct Cmd {
 }
 
 impl Cmd {
+    async fn fund_identity(&self, secret: &Secret) -> Result<(), Error> {
+        if !self.no_fund {
+            let addr = secret.public_key(self.hd_path)?;
+            let network = self.network.get(&self.config_locator)?;
+            let network_name = get_network_name(&network.rpc_url).unwrap_or(&network.rpc_url);
+            eprintln!("🌎 Funding account with public key as address on {network_name}...",);
+            match network.fund_address(&addr).await {
+                Ok(()) => {
+                    eprintln!("✅ Funded (use 'stellar keys fund me' to fund again)");
+                }
+                Err(e) => {
+                    tracing::warn!("fund_address failed: {e}");
+                }
+            }
+        }
+        Ok(())
+    }
+
+    async fn write_and_fund_identity(&self, secret: &Secret) -> Result<(), Error> {
+        self.config_locator.write_identity(&self.name, secret)?;
+        eprintln!("✅ Generated new key for '{}'", self.name);
+        eprintln!("ℹ️ Public key: {}", secret.public_key(self.hd_path)?);
+        eprintln!(
+            "ℹ️ Secret key: hidden (use 'stellar keys show {}' to view)",
+            self.name
+        );
+        self.fund_identity(secret).await?;
+        Ok(())
+    }
+
+    async fn handle_existing_identity(&self, existing_secret: &Secret) -> Result<(), Error> {
+        eprintln!("The identity {} already exists!", self.name);
+        if let Ok(root) = self.config_locator.config_dir() {
+            let mut path = root.join("identity").join(&self.name);
+            path.set_extension("toml");
+            eprintln!("    Seed phrase found at: {}", path.display());
```

Comment:
If someone attempts to generate a key for an alias that already has a key, if the command won't be doing it's primary function of generating a new key, we should have the command error with a non-zero exit status. Otherwise the command appears to have created a new-random-never-seen-before key, when in fact the alias may refer to a key that's not new and not random, which could be a security concern.

Reply:
I now throw an error `Error::IdentityAlreadyExists` which gets thrown if they try to run it without overwrite but the identity is found.